worlds/hk/__init__.py

```diff
@@ -467,25 +474,20 @@ def set_rules(self):
         elif goal == Goal.option_godhome_flower:
             multiworld.completion_condition[player] = lambda state: state.count("Godhome_Flower_Quest", player)
         elif goal == Goal.option_grub_hunt:
-            pass  # will set in stage_pre_fill()
+            multiworld.completion_condition[player] = lambda state: self.can_grub_goal(state)
         else:
             # Any goal
             multiworld.completion_condition[player] = lambda state: _hk_siblings_ending(state, player) and \
-                _hk_can_beat_radiance(state, player) and state.count("Godhome_Flower_Quest", player)
+                _hk_can_beat_radiance(state, player) and state.count("Godhome_Flower_Quest", player) and \
+                self.can_grub_goal(state)
 
         set_rules(self)
 
+    def can_grub_goal(self, state: CollectionState) -> bool:
+        return all(state.has("Grub", owner, count) for owner, count in self.grub_player_count.items())
```

Comment:
I feel like something is amiss here since it looks like both writes only assign to this dictionary but never appear to update it. So it should be the case that there is only ever one player count here unless I'm missing something

Reply:
Discussed this in DMs, seems fine